crypto/ts/index.ts

```diff
@@ -60,49 +58,16 @@ const buffer2BigInt = (b: Buffer): BigInt => {
     return snarkjs.bigInt('0x' + b.toString('hex'))
 }
 
-/* Poseidon parameters are generated from a script to meet the security requirements described in the papar.
- * Check circuits/README.md for detail.
- */
-interface PoseidonParams {
-    t: number;
-    roundFull: number;
-    roundPartial: number;
-    seed: string;
-}
-
-
-const POSEIDON_T3_PARAMS: PoseidonParams = {
-    t: 3,
-    roundFull: 8,
-    roundPartial: 49,
-    seed: POSEIDON_SEED
-}
-
-const POSEIDON_T6_PARAMS: PoseidonParams = {
-    t: 6,
-    roundFull: 8,
-    roundPartial: 50,
-    seed: POSEIDON_SEED
-}
-
-const poseidonCreateHash = (param: PoseidonParams) => {
-    return poseidon.createHash(param.t, param.roundFull, param.roundPartial, param.seed)
-}
-
 // Hash up to 2 elements
-const poseidonT3 = poseidonCreateHash(POSEIDON_T3_PARAMS)
+const poseidonT3 = (inputs: SnarkBigInt[]) => {
+    assert(inputs.length === 2)
+    return poseidon(inputs)
+}
 
 // Hash up to 5 elements
-const poseidonT6 = poseidonCreateHash(POSEIDON_T6_PARAMS)
-
-
-/*
- * A convenience function for to use mimcsponge to hash a Plaintext with
- * key 0 and require only 1 output
- */
-const hash = (plaintext: Plaintext): SnarkBigInt => {
-
-    return mimcsponge.multiHash(plaintext, 0, 1)
+const poseidonT6 = (inputs: SnarkBigInt[]) => {
+    assert(inputs.length === 5)
+    return poseidon(inputs)
 }
 
 const hash5 = (elements: Plaintext): SnarkBigInt => {
```

Comment:
We can remove `hash5` or try to add paddings in it.

Reply:
Will do!